client/my-sites/plan-features-2023-grid/hooks/use-highlight-label.ts

```diff
@@ -27,19 +26,32 @@ const useHighlightLabel = ( { planName, flowName, currentSitePlanSlug, selectedP
 	);
 	const isSuggestedPlan =
 		selectedPlan && planLevelsMatch( planName, selectedPlan ) && isAvailableForPurchase;
+	const { intent } = usePlansGridContext();
 
-	if ( flowName && isNewsletterFlow( flowName ) ) {
+	if ( 'newsletter' === intent ) {
 		if ( isPersonalPlan( planName ) ) {
 			return translate( 'Best for Newsletter' );
 		}
-	} else if ( flowName && isLinkInBioFlow( flowName ) ) {
+
+		if ( isCurrentPlan ) {
+			return translate( 'Your plan' );
+		}
```

Comment:
I was wondering if we could take this outside the if ladder, to L30. If it's the current plan, we should always return `Your plan`, right? 🤔

Reply:
I think that's correct. `your plan` should take precedence 🤔  Should move this out. 

In any case, this hook will be removed next.